k2/csrc/host/intersect.cc

```diff
@@ -52,7 +52,7 @@ void Intersection::GetSizes(Array2Size<int32_t> *fsa_size) {
   arc_map_a_.clear();
   arc_map_b_.clear();
 
-  if (IsEmpty(a_) || IsEmpty(b_)) return;
+  if (IsEmpty(a_) || IsEmpty(b_) || !check_properties_) return;
   if (!IsArcSorted(a_) || !IsArcSorted(b_)) {
```

Comment:
I would suggest

```cpp
if (check_properties_ && (!IsArcSorted(a_) || !IsArcSorted(b_)) {
  ...
}
```
The main computation of intersect is done in this function (`GetSizes`). If `check_properties_` is false,
we just skip `IsArcSorted` and go ahead.